src/main/java/org/skriptlang/skript/lang/converter/ConverterInfo.java

```diff
@@ -0,0 +1,62 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package org.skriptlang.skript.lang.converter;
+
+/**
+ * Holds information about a {@link Converter}.
+ *
+ * @param <From> The type to convert from.
+ * @param <To> The type to convert to.
+ */
+public final class ConverterInfo<From, To> {
+
+	final Class<From> from;
+	final Class<To> to;
+	final Converter<From, To> converter;
+	final int flags;
```

Comment:
Flags can be private. Use the getter. There should be no access to these fields, even at package level.